packages/react-dom/src/events/plugins/ChangeEventPlugin.js

```diff
@@ -72,7 +76,14 @@ let activeElementInst = null;
 /**
  * SECTION: handle `change` event
  */
-function shouldUseChangeEvent(elem) {
+function shouldUseChangeEvent(elem, targetInst) {
+  if (
+    enableCustomElementPropertySupport &&
+    targetInst &&
+    isCustomComponent(targetInst.elementType, targetInst.pendingProps)
```

Comment:
I think this should use memoizedProps instead which represent the ones that were committed. pendingProps represent last work in progress which could've been interrupted.

Reply:
Also, I'm able to comment out this whole line and the tests still pass. I think it's because there is no test for the `is="...."` case which `isCustomComponent` is trying to catch.

Reply:
Hmm. I understand we'd want `<div is="my-custom-element">` to work with change event.

If you do `<input is="my-custom-element">`, which branch of logic should we use? Should it use the "raw" behavior or the polyfill which we normally use for inputs?

Reply:
@sebmarkbage do you have an opinion on this? seems like it's a bit simpler to turn off polyfilling when you do `<input is="my-thing">` but that would mean that adding `is` changes semantics of your `onChange` handler. maybe that's ok because you decided to "go manual"? especially in the light of onChange->onInput future change?

Reply:
> Also, I'm able to comment out this whole line and the tests still pass. I think it's because there is no test for the is="...." case which isCustomComponent is trying to catch.

On my machine, if I comment out just the `&& isCustomComponent(...)`, 7 tests fail from various test files. If I undo the entire change here, 2 of the tests I added fail.

> Hmm. I understand we'd want `<div is="my-custom-element">` to work with change event.

Yes, `<div is='my-custom-element'>` should get this behavior as well.
I could copy all of the tests I added and use `is` instead of `<my-custom-element />` to show that I used `isCustomComponent`, but you could say the same thing about all of the many custom element tests I added...

> But if you do <input is="my-custom-element">, which branch of logic should we use? Should it use the "raw" behavior or the polyfill which we normally use for inputs?

Based on what I read in DOMPluginEventSystem, ChangeEventPlugin is a "polyfill", right? The code I added is in ChangeEventPlugin, which means that `<input>` and custom elements should now be treated the same with regards to this behavior, right...?

> I think this should use memoizedProps instead which represent the ones that were committed. pendingProps represent last work in progress which could've been interrupted.

Thanks, I was wondering which I should use! I have changed it to memoizedProps.

Reply:
>Based on what I read in DOMPluginEventSystem, ChangeEventPlugin is a "polyfill", right? The code I added is in ChangeEventPlugin, which means that <input> and custom elements should now be treated the same with regards to this behavior, right...?

I meant the code of the polyfill below (look how this function is used). In the polyfill itself, we decide between using one of the events under the hood: `change`, `input`, or `click`. We decide that based on the type of the element. The code you added makes us always use `change` event for custom elements. That's good for `<custom-element>` or `<div is="custom-element">`. What I'm not sure about is `<input is="custom-element">`. Should it use the `change` event (which means it would work differently from `<input>` which in React maps `onChange` to `input`)? Or should it use the `input`/`click` events to mirror how `<input>` works? And then there'd still be event mapping for some custom elements?

Using raw events for `<input is="custom-element">` feels like simpler behavior to me but I'm not sure. There's also a potential intent to change it for `<input>` in distant future altogether: https://github.com/facebook/react/issues/9657.

Reply:
@josepharhar Taking a step back, we should look at how controlled inputs generally work with `<input is="my-custom-element">`. Like how are you supposed to write a controlled input? One of the features of controlled inputs is that `<input value="">` with no `onChange` handler won't allow typing. The same should be true for `<input value="" is="my-custom-element">`. Is that the case? If it already works, is it necessary to add `onChange` (like normal React) or `onInput` (closer to browser behavior) to make it respond to typing? That's what we need to decide on.

Reply:
I just talked a bit with some LitElement folks about this, and they pointed out that Safari doesn't support the `is` attribute at all and that they advise against using it: https://bugs.webkit.org/show_bug.cgi?id=182671

I think that it would make the most sense to continue doing all of react's current behavior for form controls when they have the `is` attribute. Besides, the element and its shadowdom is still the same in this case when using `is`.

I pushed a commit to do this with tests.